ckanext/hierarchy/templates/package/base.html

```diff
@@ -12,9 +12,9 @@
             {% endfor %}
             <li>{% link_for organization|truncate(30), controller='organization', action='read', id=pkg.organization.name %}</li>
         {% else %}
-            <li>{% link_for _('Datasets'), controller='package', action='search' %}</li>
+            <li>{% link_for _('Datasets'), controller='dataset' if h.ckan_version() > '2.9' else 'package', action='search' %}</li>
```

Comment:
This check will fail with 2.10 (master):

```suggestion
            <li>{% link_for _('Datasets'), controller='dataset' if h.ckan_version().split('.')[1] | int >= 9 else 'package', action='search' %}</li>
```